### PR TITLE
KAZOO-4398 - Generate recording_url

### PR DESCRIPTION
**Pull request by pramsky**
Generates recording url using media_transfer_destination and media_file
  from the channel_vars in the cdr.
 This displays the recording_url when called via API, or exported to csv.
 Thanks to @fenollp for the suggestions

### CODE REVIEW

**Comment by fenollp** [comment]
Follow up of #1453

File: applications/crossbar/src/modules/cb_cdrs.erl [hunk 1]
```diff
@@ -520,12 +520,24 @@ col_account_call_type(JObj, _Timestamp) -> wh_json:get_value([<<"custom_channel_
...
 col_reseller_cost(JObj, _Timestamp) -> wh_util:to_binary(reseller_cost(JObj)).
 col_reseller_call_type(JObj, _Timestamp) -> wh_json:get_value([<<"custom_channel_vars">>, <<"reseller_billing">>], JObj, <<>>).
 
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% generates a recording_url entry from media_transfer_destination and media_file
+%% @end
+%%--------------------------------------------------------------------
+-spec recording_url(api_binary(), api_binary()) -> api_binary().
```
Comment (fenollp) [inline applications/crossbar/src/modules/cb_cdrs.erl]: Spec should be `-spec recording_url(ne_binary(), ne_binary()) -> api_binary().`

Comment (pramsky) [inline applications/crossbar/src/modules/cb_cdrs.erl]: Got it. I'll updated and issue another pull request.

**Comment by fenollp** [comment]
wait sorry not `ne_binary()` nor `api_binary()` but `binary()`. I was very tired…

**Comment by k-anderson** [comment]
@fenollp what was recording_url? Please ensure this is a backward compatible change, if not we will need to make it a new parameter in the result set so as to not disrupt those who might have already coded to the current value of recording_url.

**Comment by fenollp** [comment]
https://github.com/2600hz/kazoo/commit/9925c4c39291d7b55f054a4aa611587640de2610
@k-anderson not sure how to make sure this doesn't break something the same way above commit might have broken stuff too..

@pramsky Did you try using the code in the commit I mention? What is your experience with it?

**Comment by pramsky** [comment]
@fenollp 
With the earlier commit, in the cdrs the recording_url was always blank. 

I could go into the details of the cdr and see the media_transfer_destination and media_file for recording in there. The recording_url was not populated in the API result or in the exported csv.

**Comment by k-anderson** [comment]
@pramsky howdy!  Unfortunately as we progress into 4.0 this is no longer a valid issue as media_transfer_destination does not exist anymore.  However, as this is existing functionality I have asked @lazedo to ensure this will be available with the new recording process.  Please watch the announcements because the new recording process allows multiple recordings per call and as such we may change the recording_url to an array!

https://github.com/2600hz/kazoo/blob/master/ANNOUNCEMENTS.md

**Comment by pramsky** [comment]
@k-anderson 

Thanks for the update =) Looking forward to 4 =)

**Pull request closed without merging**